server/build/webpack.js

```diff
@@ -246,7 +246,7 @@ export default async function getBaseWebpackConfig (dir, {dev = false, isServer
         }
       }),
       new webpack.DefinePlugin({
-        'process.env.NODE_ENV': JSON.stringify(dev ? 'development' : 'production')
+        'process.env.NODE_ENV': JSON.stringify(process.env.NODE_ENV ? process.env.NODE_ENV : defaultNodeEnv)
```

Comment:
This actually breaks a ton of packages when combined with uglify and suppresses development warning in React when in dev mode.

If you want to expose something to the client side you'll be able to use https://github.com/zeit/next.js/pull/3882 when it lands (today)

Reply:
Looks cool